config/v1/types_cluster_operator.go

```diff
@@ -49,13 +51,20 @@ type ClusterOperatorStatus struct {
 	// versions is a slice of operator and operand version tuples.  Operators which manage multiple operands will have multiple
 	// operand entries in the array.  Available operators must report the version of the operator itself with the name "operator".
 	// An operator reports a new "operator" version when it has rolled out the new version to all of its operands.
+	// +listType=map
+	// +listMapKey=name
 	// +optional
 	Versions []OperandVersion `json:"versions,omitempty"`
 
 	// relatedObjects is a list of objects that are "interesting" or related to this operator.  Common uses are:
 	// 1. the detailed resource driving the operator
 	// 2. operator namespaces
 	// 3. operand namespaces
+	// +listType=map
+	// +listMapKey=group
+	// +listMapKey=resource
+	// +listMapKey=namespace
```

Comment:
Adding `namespace` here results in it becoming required, which it should not be. However, if present it should be unique, and an entry where it is not present is distinct from one where it is.

This is ugly. I'm going to punt on it and just remove it because I don't need it myself.